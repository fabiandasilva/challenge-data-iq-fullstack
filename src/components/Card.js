import Image from "next/image";
import Link from "next/link";

import { Email, Location, Phone, Work } from "../../public/image/icons/index";

const Card = ({ user }) => {
  return (
    <div className="mb-24 md:mb-0">
      <div className="block h-full rounded-lg bg-white shadow-[0_2px_15px_-3px_rgba(0,0,0,0.07),0_10px_20px_-2px_rgba(0,0,0,0.04)]">
        <div className="flex justify-center">
          <div className="flex justify-center -mt-[75px]">
            <Image
              src={user.pictureUrl.large}
              className="mx-auto rounded-full shadow-lg  w-[150px]"
              alt={user.name}
              width="110"
              height="110"
            />
          </div>
        </div>
        <div className="px-6 pb-6 pt-2">
          <h5 className="text-xl font-bold">{user.name}</h5>
          <p className="text-lg pb-4">@{user.username}</p>
          <div className="flex items-center gap-2">
            <Image
              src={Work}
              width="30"
              height="20"
              alt="icon"
              className="h-5 w-5"
            />
            <Link
              href={`https://www.google.com/search?q=${user.company.name}`}
              rel="noopener noreferrer"
              target="_blank"
            >
              <span className="text-zinc-700 cursor-pointer">
                {user.company.name}
              </span>
            </Link>
          </div>
          <div className="flex items-center gap-2">
            <Image
              src={Location}
              width="20"
              height="20"
              alt="icon"
              className="h-5 w-5"
            />
            <Link
              href={`https://www.google.com.ar/maps/search/${user.address.city}`}
              rel="noopener noreferrer"
              target="_blank"
            >
              <span className="text-zinc-700 cursor-pointer">
                {user.address.city}
              </span>
            </Link>
          </div>
          <div className="flex items-center gap-2">
            <Image
              src={Email}
              width="20"
              height="20"
              alt="icon"
              className="h-5 w-5"
            />
            <Link
              href={`mailto:${user.email}`}
              rel="noopener noreferrer"
              target="_blank"
            >
              <span className="text-zinc-700 cursor-pointer">{user.email}</span>
            </Link>
          </div>
          <div className="flex items-center gap-2">
            <Image
              src={Phone}
              width="20"
              height="20"
              alt="icon"
              className="h-5 w-5"
            />
            <span className="text-zinc-700 cursor-pointer">{user.phone}</span>
          </div>

          <div className="flex items-center justify-center pt-6">
            <Link href={`/${user.id}`}>
              <button className="bg-[#6D8A96] hover:bg-[#a0b3ba] text-white text-sm font-bold py-2 px-4 rounded">
                MÃ¡s info
              </button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
