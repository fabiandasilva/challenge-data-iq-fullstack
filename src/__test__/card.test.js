/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import React from "react";
import Card from "../components/Card";
import user from "./mock/mockUser";

test("Card se renderiza correctamente", () => {
  const { asFragment } = render(<Card user={user} />);

  expect(asFragment()).toMatchSnapshot();
});
