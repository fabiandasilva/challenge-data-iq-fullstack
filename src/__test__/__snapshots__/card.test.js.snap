// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card se renderiza correctamente 1`] = `
<DocumentFragment>
  <div
    class="mb-24 md:mb-0"
  >
    <div
      class="block h-full rounded-lg bg-white shadow-[0_2px_15px_-3px_rgba(0,0,0,0.07),0_10px_20px_-2px_rgba(0,0,0,0.04)]"
    >
      <div
        class="flex justify-center"
      >
        <div
          class="flex justify-center -mt-[75px]"
        >
          <img
            alt="John Doe"
            class="mx-auto rounded-full shadow-lg  w-[150px]"
            data-nimg="1"
            decoding="async"
            height="110"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=256&q=75 2x"
            style="color: transparent;"
            width="110"
          />
        </div>
      </div>
      <div
        class="px-6 pb-6 pt-2"
      >
        <h5
          class="text-xl font-bold"
        >
          John Doe
        </h5>
        <p
          class="text-lg pb-4"
        >
          @johndoe
        </p>
        <div
          class="flex items-center gap-2"
        >
          <img
            alt="icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="20"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
          <a
            href="https://www.google.com/search?q=Example Company"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="text-zinc-700 cursor-pointer"
            >
              Example Company
            </span>
          </a>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <img
            alt="icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="20"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="20"
          />
          <a
            href="https://www.google.com.ar/maps/search/Example City"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="text-zinc-700 cursor-pointer"
            >
              Example City
            </span>
          </a>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <img
            alt="icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="20"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="20"
          />
          <a
            href="mailto:johndoe@example.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="text-zinc-700 cursor-pointer"
            >
              johndoe@example.com
            </span>
          </a>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <img
            alt="icon"
            class="h-5 w-5"
            data-nimg="1"
            decoding="async"
            height="20"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="20"
          />
          <span
            class="text-zinc-700 cursor-pointer"
          >
            123-456-7890
          </span>
        </div>
        <div
          class="flex items-center justify-center pt-6"
        >
          <a
            href="/1"
          >
            <button
              class="bg-[#6D8A96] hover:bg-[#a0b3ba] text-white text-sm font-bold py-2 px-4 rounded"
            >
              MÃ¡s info
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
